lib/internal/test_runner/runner.js

```diff
@@ -161,29 +166,20 @@ class FileTest extends Test {
           directive = kEmptyObject;
         }
 
-        if (pass) {
-          this.reporter.ok(
-            nesting,
-            this.name,
-            node.id,
-            node.description,
-            YAMLToJs(node.diagnostics),
-            directive
-          );
-        } else {
-          this.reporter.fail(
-            nesting,
-            this.name,
-            node.id,
-            node.description,
-            YAMLToJs(node.diagnostics),
-            directive
-          );
+        this.#hasSubtests = true;
+        const diagnostics = YAMLToJs(node.diagnostics);
+        const cancelled = diagnostics.error?.failureType === kCancelledByParent;
+        const testNumber = nesting === 0 ? (Number(node.id) + this.testNumber - 1) : node.id;
+        const method = pass ? 'ok' : 'fail';
+        this.reporter[method](nesting, this.name, testNumber, node.description, diagnostics, directive);
+        if (nesting === 0) {
+          this.parent.countSubtest({ finished: true, skipped: skip, isTodo: todo, passed: pass, cancelled });
+          this.countSubtest({ finished: true, skipped: skip, isTodo: todo, passed: pass, cancelled });
```

Comment:
Since this is the CLI runner, do we actually need to track the subtests for the current test and the root? I would think we only need one (likely the root).